### PR TITLE
cmd/cloner,*: optimize nillable slice cloner

### PR DESCRIPTION
**Pull request by raggi**
A wild @josharian appears with a good suggestion for a refactor, thanks Josh!

Updates #9410
Signed-off-by: James Tucker <james@tailscale.com>

### CODE REVIEW

File: net/dnsfallback/dns-fallback-servers.json [hunk 1]
```diff
@@ -6,25 +6,33 @@
 			"RegionName": "r1",
 			"Nodes": [
 				{
-					"Name": "1c",
+					"Name": "1f",
```
Comment (maisem) [inline net/dnsfallback/dns-fallback-servers.json:9]: revert this?

Comment (raggi) [reply net/dnsfallback/dns-fallback-servers.json]: 👍🏼   https://github.com/tailscale/tailscale/pull/9602

File: cmd/cloner/cloner.go [hunk 2]
```diff
@@ -122,8 +122,7 @@ func gen(buf *bytes.Buffer, it *codegen.ImportTracker, typ *types.Named) {
 		case *types.Slice:
 			if codegen.ContainsPointers(ft.Elem()) {
 				n := it.QualifiedName(ft.Elem())
-				writef("if src.%s != nil {", fname)
-				writef("dst.%s = make([]%s, len(src.%s))", fname, n, fname)
+				writef("dst.%s = append([]%s(nil), make([]%s, len(src.%s))...)", fname, n, n, fname)
```
Comment (maisem) [inline cmd/cloner/cloner.go:125]: doesn't this end up allocating two slices? does the compiler optimize it away?

cc @josharian

Comment (josharian) [reply cmd/cloner/cloner.go:125]: See https://go-review.googlesource.com/c/go/+/109517 :)

And for lots more discussion, https://github.com/golang/go/issues/24204 and https://github.com/golang/go/issues/21266

Comment (maisem) [reply cmd/cloner/cloner.go:125]: thanks!

**Review by maisem — APPROVED** [review]

**Pull request merged**